### DISCUSSION TITLE
Working with local apps

### DISCUSSION THREAD

**Discussion by nicovoni** [SafeExamBrowser/seb-win-refactoring#1001] [Q&A]
I know SEB is mainly used for online resources but I wonder, is it possible to open a local app inside SEB?
I need students to work with a software on the PC and then send me the file. Ideas?

**Comment by dbuechel** [comment]
You can (if required) completely disable the internal browser engine (see https://safeexambrowser.org/windows/win_usermanual_en.html#BrowserPane > Use SEB without browser window) and then configure permitted applications according to your requirements (see https://safeexambrowser.org/windows/win_usermanual_en.html#ApplicationsPane).

**Reply by nicovoni** [reply]
Instead, is it possible to use SEB with its browser window but also allowing you to work for example with notepad, save the file and upload it on a web page opened in the SEB browser window?

**Reply by dbuechel** [reply]
Certainly, as is explained in the user manual linked above.

**Reply by nicovoni** [reply]
@dbuechel I did some tests and I ran into this problem. The students work with an IDE that I added to the allowed applications. The problem arises when the executable file (result of the compilation) cannot be opened because it is not among the allowed applications. Is it possible to allow a program even if it does not exist yet? For example, I could tell the students to call the program pippo.exe

**Reply by dbuechel** [reply]
Yes, that should work. You can easily verify this yourself by enabling the real-time application log (see the security tab in the configuration tool) and trying the use case yourself.

**Reply by nicovoni** [reply]
Thanks, could you please explain it to me practically? my use case is: allow all executable applications whose name starts with "class1a".

**Reply by dbuechel** [reply]
That is unfortunately not possible, we do not support wildcards or alike for the configuration of third-party applications. You'll need to configure all applications you want to allow specifically.

**Reply by nicovoni** [reply]
Okay, thank you. I have another question related to local applications: students have to work on HTML code with notepad++, then they have to "test" their HTML file. If they open the HTML file with a browser outside SEB (chrome, edge, firefox), I have no ability to control it (they can navigate to other websites). Is it possible to open HTML files with SEB's internal browser?

**Reply by dbuechel** [reply]
Yes, that should certainly be possible using a local file URI (e.g. `file:///C:/...`).
